packages/story-editor/src/components/form/tags/tag.js

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import {
+  Icons,
+  Text,
+  themeHelpers,
+  THEME_CONSTANTS,
+  Tooltip,
+  TOOLTIP_PLACEMENT,
+} from '@web-stories-wp/design-system';
+import { __ } from '@web-stories-wp/i18n';
+import { useMemo } from '@web-stories-wp/react';
+import PropTypes from 'prop-types';
+import styled, { css } from 'styled-components';
+import { v4 as uuidv4 } from 'uuid';
+
+const Dismiss = styled.button`
+  all: unset;
+  cursor: pointer;
+  border-radius: ${({ theme }) => theme.borders.radius.small};
+  ${themeHelpers.focusableOutlineCSS};
+  width: 32px;
+  min-width: 32px;
+  height: 32px;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+
+  svg {
+    height: 16px;
+    width: 16px;
+    margin: auto;
+  }
+`;
+
+const Token = styled.span`
+  ${({ theme }) => css`
+    color: ${theme.colors.fg.primary};
+    border: 1px solid ${theme.colors.border.defaultNormal};
+    border-radius: ${theme.borders.radius.small};
+  `}
+  position: relative;
+  display: flex;
+  align-items: center;
+  padding: 3px 2px;
+  margin: 3px 5px 3px 0;
+  max-width: calc(100% - 16px);
+`;
+
+const TokenText = styled(Text).attrs({
+  forwardedAs: 'span',
+  size: THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL,
+})`
+  padding-left: 11px;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+`;
+
+function Tag({ children, onDismiss }) {
+  const id = useMemo(() => uuidv4(), []);
+  return (
+    <Token>
+      <TokenText id={id}>{children}</TokenText>
+      <Tooltip
+        title={__('Remove Tag', 'web-stories')}
+        placement={TOOLTIP_PLACEMENT.BOTTOM}
+        hasTail
+      >
+        <Dismiss
+          onClick={onDismiss}
+          aria-label={__('Remove Tag', 'web-stories')}
```

Comment:
It'd be great if this specified what tag was getting removed since a screenreader doesn't read the spans.

Reply:
So gutenberg does it like this:
<img width="389" alt="Screen Shot 2021-09-03 at 2 07 25 PM" src="https://user-images.githubusercontent.com/35983235/132054955-156202b4-eea7-491f-a714-3bff6c4dcf14.png">

Which is what we're doing here. I think it might be fine because we pass it a `aria-describedby` attribute.

Reply:
trueee, if it's good for Gutenberg it's fine for us. 🤝